### PR TITLE
Feature MSRP pruning

### PR DESCRIPTION
**Pull request by andrew-elder**
Add support for optionally pruning TalkerAdvertise and TalkerFailed attributes from those actively maintained by mrpd.

### CODE REVIEW

**Comment by andrew-elder** [comment]
Compiler failure - oops.

**Comment by andrew-elder** [comment]
All, this pull request is now ready for review. Comments welcomed! Thanks.

File: daemons/mrpd/msrp.c [hunk 1]
```diff
@@ -3222,6 +3277,20 @@ int msrp_dumptable(struct sockaddr_in *client)
...
 
+static int msrp_cmd_parse_stream_id(char *buf, int buflen,
+                                    uint8_t *stream_id,
+				    int *err_index )
+{
+	struct parse_param specs[] = {
+		{"S" PARSE_ASSIGN, parse_c64, stream_id },
+		{0, parse_null, 0 }
+	};
+	if( buflen < 16+4 )
```
Comment (pinealservo) [inline daemons/mrpd/msrp.c]: Might be good to explain the magic numbers here and below.

File: daemons/mrpd/msrp.c [hunk 2]
```diff
@@ -3625,8 +3790,18 @@ int msrp_init(int msrp_enable)
 	if (NULL == MSRP_db)
 		goto abort_socket;
 
+	if( max_interesting_stream_ids < 8 ) {
+		/* Allow minimum 8 interesting talker stream ids and 8 interesting listener stream ids */
```
Comment (pinealservo) [inline daemons/mrpd/msrp.c]: We're only really tracking talker ids, right?

File: daemons/mrpd/msrp.c (hunk 1 above)
Comment (andrew-elder) [inline daemons/mrpd/msrp.c]: Agreed. Will fix.

File: daemons/mrpd/msrp.c (hunk 2 above)
Comment (andrew-elder) [inline daemons/mrpd/msrp.c]: Yes, talkerIDs are what the max_interesting applies to. So if you are suggesting I rename the  max_interesting_stream_ids variable, I agree.

Comment (pinealservo) [inline daemons/mrpd/msrp.c]: I was actually referring to the comment, which suggests that there are 16 total stream_ids, 8 talker and 8 listener.

**Comment by pinealservo** [comment]
I read through all the code, and it looks good to me. I like the tests, though I don't have the time to dig too deeply into how they're working. It looks like they provide reasonable coverage of the changes, though.

**Comment by davidcemin** [comment]
I did read through all the code as well and all looks good.

**Comment by pinealservo** [comment]
OK, looks like all the open comments were resolved.

**Pull request merged**